### PR TITLE
First cut of a complete "what are entities?"

### PR DESCRIPTION
**Pull request by jyucsiro**
Documentation page for explaining/documenting the background behind "entities" or registered items in the M&DS IS. 

This page is aimed at a general audience who wants to understand what are "entities" and why it is necessary to register them. It covers a defined list of "types" for M&DS IS, and a first attempt at describing them. The content is more descriptive rather than a "how-to" doc. It provides context for the "how-to" pages...

Welcome suggestions/feedback.

### CODE REVIEW

**Comment by jyucsiro** [comment]
There is also this page: https://gbrrestoration.github.io/rrap-mds-knowledge-hub/information-system/provenance/overview/what-is-provenance.html

Which is an light intro to provenance and why we need it in the M&DS IS

File: information-system/provenance/overview/what-are-entities.md [hunk 1]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
...
+The specialised types used in M&DS IS are provided below:
+
+| Type        | Sub-type           | Description  |
+| ------------- |-------------| -----|
+| Entity      | Dataset Template | A defined template of a dataset and its anticipated set of resources. |
+| Entity      | Dataset | Structured collection of data/files associated with a body of work. |
+| Entity      | Model Run Workflow Template |  A defined template of a model run workflow and its associated set of input and output templates, and annotations. |
+| Agent      | Organisation | "An organization is a social or legal institution such as a company, society, etc." [1] |
+| Agent      | Person | "Person agents are people" [1] |
+| Activity      | Model Run | An computational modelling activity that estimates, simulates and/or calculates an outcome based on a model (simplification or abstraction of a perceived real world phenomena), and defined processes and inputs (e.g. datasets, and model parameters). |
```
Comment (PeterBaker0) [inline information-system/provenance/overview/what-are-entities.md]: Just noting "An" instead of "A"

**Comment by PeterBaker0** [comment]
Reads very well to me - good job

File: information-system/provenance/overview/what-are-entities.md [hunk 2]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
 ---
 # What are entities?
 
-In the RRAP M&DS Information System (IS), entities are things that are registered in the M&DS IS Registry. Registering entities is fundamental to establishing stable provenance. 
+## 1. Overview
 
-Each registered item (entity) is given a unique and persistent identifier (ID). These IDs provide a mechanism for 2 things:
+In the RRAP M&DS Information System (IS), entities are things that are registered in the M&DS IS Registry. Registration of entities is fundamental to establishing stable provenance. It establishes a record for each entity used or generated in producing or delivering an artifact in a provenance record.
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md]: 'registered' --> 'recorded' ?

File: information-system/provenance/overview/what-are-entities.md [hunk 3]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
 ---
 # What are entities?
 
-In the RRAP M&DS Information System (IS), entities are things that are registered in the M&DS IS Registry. Registering entities is fundamental to establishing stable provenance. 
+## 1. Overview
 
-Each registered item (entity) is given a unique and persistent identifier (ID). These IDs provide a mechanism for 2 things:
+In the RRAP M&DS Information System (IS), entities are things that are registered in the M&DS IS Registry. Registration of entities is fundamental to establishing stable provenance. It establishes a record for each entity used or generated in producing or delivering an artifact in a provenance record. 
+
+In the RRAP M&DS IS, each registered item (entity) is given a unique and persistent identifier (ID). These IDs provide a mechanism for 2 things:
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md]: Allocation of an identifier signifies successful registration.

File: information-system/provenance/overview/what-are-entities.md [hunk 4]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
...
 2. Entities are able to link to other entities (i.e. object references rather than just text descriptions)
 
-These registered things can then be referred to in provenance records, e.g. a provenance record about a model run is able to link to defined descriptions of the model that was used, who ran the model, a list of input data, and the outputs/results. The ability to link to entities reduced manual error and duplication of descriptions, as they can be reused and simply referred to. Registering provenance records with the entity references provides a way to create provenance graphs. These provenance graphs can then be queried. 
+In addition to this, the persistent IDs for each entity in the RRAP M&DS IS is able to be hyperlinked and looked up in the IS via the web.
+
+These registered entities can then referenced in provenance records, e.g. a provenance record about a model run references a computational model that was used, the agent or person who ran the model, a list of input datasets, and the output dataset/results. The ability to reference entities reduces manual error and duplication of descriptions, as they can be reused and simply referred to.
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md]: 'references' --> 'refers to' ?

File: information-system/provenance/overview/what-are-entities.md [hunk 5]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
...
 
-In the RRAP M&DS Information System (IS), entities are things that are registered in the M&DS IS Registry. Registering entities is fundamental to establishing stable provenance. 
+## 1. Overview
 
-Each registered item (entity) is given a unique and persistent identifier (ID). These IDs provide a mechanism for 2 things:
+In the RRAP M&DS Information System (IS), entities are things that are registered in the M&DS IS Registry. Registration of entities is fundamental to establishing stable provenance. It establishes a record for each entity used or generated in producing or delivering an artifact in a provenance record. 
+
+In the RRAP M&DS IS, each registered item (entity) is given a unique and persistent identifier (ID). These IDs provide a mechanism for 2 things:
 1. It allows information to be attached to each entity in a defined way
 2. Entities are able to link to other entities (i.e. object references rather than just text descriptions)
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md]: 'references' --> 'cross-references'?

File: information-system/provenance/overview/what-are-entities.md [hunk 6]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
...
+In addition to this, the persistent IDs for each entity in the RRAP M&DS IS is able to be hyperlinked and looked up in the IS via the web.
+
+These registered entities can then referenced in provenance records, e.g. a provenance record about a model run references a computational model that was used, the agent or person who ran the model, a list of input datasets, and the output dataset/results. The ability to reference entities reduces manual error and duplication of descriptions, as they can be reused and simply referred to. 
+
+Registering provenance records with references to the registered entity provides the IS with the ability to create provenance graphs. Users of the IS are then able to explore and query these provenance graphs.
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md]: 'with references' --> 'using cross-references' ?

Comment (dr-shorthair) [reply information-system/provenance/overview/what-are-entities.md]: Is 'provenance graph' defined anywhere?

File: information-system/provenance/overview/what-are-entities.md [hunk 7]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
...
+* Agent
+* Activity
+
+![PROV-O Conceptual Model](https://upload.wikimedia.org/wikipedia/commons/thumb/7/7f/W3C_PROV_Data_Model.svg/661px-W3C_PROV_Data_Model.svg.png "PROV-O Conceptual Model")
+
+"An entity is a physical, digital, conceptual, or other kind of thing with some fixed aspects; entities may be real or imaginary" [1]. An *Entity* represents a thing in the world. An entity may be derived from other entities, and may be generated by an *Activity* that used a set of entities. 
+
+An *Activity* "is something that occurs over a period of time and acts upon or with entities; it may include consuming, processing, transforming, modifying, relocating, using, or generating entities". [1]
+
+"An *Agent* is something that bears some form of responsibility for an activity taking place, for the existence of an entity, or for another agent's activity [1]. This may be a person or a piece of software that is executed and associated with the activity that generates the Entity. It provides a way to attribute the activity and generation of the Entity to that agent.
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md]: Agent could be considered as a sub-class of Entity - that is not inconsistent with the PROV model.

Else 'Party' is sub-class of Entity, and 'Agent' a subclass of Party, being those Parties who have responsibility (agency) for an Activity.

File: information-system/provenance/overview/what-are-entities.md [hunk 8]
```diff
@@ -8,14 +8,58 @@ parent: Provenance Overview
...
+
+The specialised types used in M&DS IS are provided below:
+
+| Type        | Sub-type           | Description  |
+| ------------- |-------------| -----|
+| Entity      | Dataset Template | A defined template of a dataset and its anticipated set of resources. |
+| Entity      | Dataset | Structured collection of data/files associated with a body of work. |
+| Entity      | Model Run Workflow Template |  A defined template of a model run workflow and its associated set of input and output templates, and annotations. |
+| Agent      | Organisation | "An organization is a social or legal institution such as a company, society, etc." [1] |
+| Agent      | Person | "Person agents are people" [1] |
```
Comment (dr-shorthair) [inline information-system/provenance/overview/what-are-entities.md:68]: What about software agents?

File: information-system/provenance/overview/what-are-entities.md (hunk 6 above)
Comment (jyucsiro) [reply information-system/provenance/overview/what-are-entities.md]: It isn't, but i've added one in with a diagram

File: information-system/provenance/overview/what-are-entities.md (hunk 8 above)
Comment (jyucsiro) [reply information-system/provenance/overview/what-are-entities.md:68]: Hmm, we haven't defined a subtype called "software agents" yet. I think we'd have one down the track as users start to have software agents, but I haven't seen a pull for it yet. @PeterBaker0 - have you had any conversations about this yet with users?

File: information-system/provenance/overview/what-are-entities.md (hunk 7 above)
Comment (jyucsiro) [reply information-system/provenance/overview/what-are-entities.md]: I'm leaning more towards "Agent could be considered as a sub-class of Entity". The latter while possible, is a bit more complex. Not sure we need that level of semantics in our system yet

File: information-system/provenance/overview/what-are-entities.md (hunk 2 above)
Comment (dr-shorthair) [reply information-system/provenance/overview/what-are-entities.md]: Are model runs also registered? Need to be careful about use of term 'Entities'. `prov:Entity` is a continuant. do you intend that registered entities include both Continuants and Occurrents?

Comment (jyucsiro) [reply information-system/provenance/overview/what-are-entities.md]: yes, we're mixing the definition of prov:Entity and entities in the everyday general sense... i had raised this issue with @PeterBaker0 

agree the semantics are different if we're including registration of Occurrents (i.e. Activity > Model Run). 

however, at the time, it was felt that entities in the everyday general sense was sufficient. maybe not. what would be an alternative? I suggested "Thing" but that might be too general... Is there a suitable alternative?

Comment (jyucsiro) [reply information-system/provenance/overview/what-are-entities.md]: We could substitute like this:
- Entity -> Registered item
- Entity registry -> Registry

Comment (dr-shorthair) [reply information-system/provenance/overview/what-are-entities.md]: I just helped Peter with some registry description words. 
We used 'Thing'.

Comment (dr-shorthair) [reply information-system/provenance/overview/what-are-entities.md]: (Interesting that we are re-discovering the ISO 19135 model that was implemented in LDR ...)

File: information-system/provenance/overview/what-are-entities.md (hunk 8 above)
Comment (PeterBaker0) [reply information-system/provenance/overview/what-are-entities.md:68]: No - I think the main pull has been for the ability to more specifically attribute responsibility/credit e.g. 'collaborators'.

**Comment by PeterBaker0** [comment]
I'm going to merge this - we can discuss further changes if required.

**Comment by PeterBaker0** [comment]
I'll leave the branch here for now

**Pull request merged**